esphome/components/dallas_temp/sensor.py

```diff
@@ -0,0 +1,44 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import one_wire, sensor
+from esphome.const import (
+    CONF_RESOLUTION,
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_CELSIUS,
+)
+
+dallas_temp_ns = cg.esphome_ns.namespace("dallas_temp")
+
+DallasTemperatureSensor = dallas_temp_ns.class_(
+    "DallasTemperatureSensor",
+    cg.PollingComponent,
+    sensor.Sensor,
+    one_wire.OneWireDevice,
+)
+
+CONFIG_SCHEMA = (
+    sensor.sensor_schema(
+        DallasTemperatureSensor,
+        unit_of_measurement=UNIT_CELSIUS,
+        accuracy_decimals=1,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    )
+    .extend(
+        {
+            cv.Optional(CONF_RESOLUTION, default=12): cv.int_range(min=9, max=12),
+        }
+    )
+    .extend(one_wire.one_wire_device_schema())
+    .extend(cv.polling_component_schema("60s"))
+)
+
+
+async def to_code(config):
+    var = await sensor.new_sensor(config)
+    await cg.register_component(var, config)
+    await one_wire.register_one_wire_device(var, config)
+
+    if CONF_RESOLUTION in config:
+        cg.add(var.set_resolution(config[CONF_RESOLUTION]))
```

Comment:
There is a default so checking is not needed
```suggestion
    cg.add(var.set_resolution(config[CONF_RESOLUTION]))
```